index.js

```diff
@@ -49,65 +68,91 @@ Rollout.prototype.handler = function (key, flags) {
+      if (err) {
+        return reject(err)
+      }
       resolve(result)
     })
   })
-  .then(Promise.all.bind(Promise, promises))
+  .then(function () {
+    return Promise.all(promises)
+  })
 }
 
 Rollout.prototype.get = function (key, id, opt_values, multi) {
   var flags = this._handlers[key]
   var likely = this.val_to_percent(key + id)
-  var _id = {
-    id: id
-  }
-  if (!opt_values) opt_values = _id
-  if (!opt_values.id) opt_values.id = id
-  var keys = Object.keys(flags).map(function (k) {
-    return key + ':' + k
+  var _id = { id: id }
+  opt_values = opt_values || _id
+  opt_values.id = opt_values.id || id
+  var keys = Object.keys(flags).map(function (mod) {
+    return key + ':' + mod
   })
   var client = multi || this.client
   return new Promise(function (resolve, reject) {
     client.mget(keys, function (err, result) {
-      if (err) return reject(err)
+      if (err) {
+        return reject(err)
+      }
       resolve(result)
     })
   })
   .then(function (percentages) {
     var i = 0
     var deferreds = []
+    var output
     for (var modifier in flags) {
       // in the circumstance that the key is not found, default to original value
       if (percentages[i] === null) {
-        percentages[i] = flags[modifier].percentage
+        percentages[i] = normalizePercentageRange(flags[modifier].percentage)
       }
-      if (likely < percentages[i]) {
-        if (!flags[modifier].condition) flags[modifier].condition = defaultCondition
-        var output = flags[modifier].condition(opt_values[modifier])
+      if (isPercentageInRange(likely, percentages[i])) {
+        if (!flags[modifier].condition) {
+          flags[modifier].condition = defaultCondition
+        }
+        output = flags[modifier].condition(opt_values[modifier])
         if (output) {
-          if (typeof output.then === 'function') deferreds.push(output)
-          else return true
+          if (typeof output.then === 'function') {
+            // Normalize thenable to Bluebird Promise
+            // Reflect the Promise to coalesce rejections
+            output = Promise.resolve(output).reflect()
+            output.flagModifier = modifier
+            deferreds.push(output)
+          } else {
+            return modifier
+          }
         }
       }
       i++
     }
     if (deferreds.length) {
       return Promise.all(deferreds)
       .then(function (results) {
+        var resultPromise, resultValue
         for (var i = 0, len = results.length; i < len; i++) {
-          if (results[i]) return true
+          resultPromise = results[i]
+          // Treat rejected conditions as inapplicable modifiers
+          if (resultPromise.isFulfilled()) {
+            resultValue = resultPromise.value()
+            // Treat resolved conditions with non-false values as affirmative
+            // (This is to handle `Promise.resolve()` and `Promise.resolve(null)`)
+            if (resultValue !== false) {
```

Comment:
this seems to conflict with changes in #6

Reply:
In #6 `false` and `null` are rejected, but arguably only `false` should be rejected because Bluebird recommends resolving Promises with `null` instead of `undefined` to silence the ["a promise was created in a handler but was not returned from it" warning](http://bluebirdjs.com/docs/warning-explanations.html#warning-a-promise-was-created-in-a-handler-but-was-not-returned-from-it)

Since we're doing breaking API changes anyway, I'll change this to require an explicitly truthy resolved value instead of relying on vagaries.